web3/gas_strategies/time_based.py

```diff
@@ -19,10 +19,10 @@
 def _get_avg_block_time(w3, sample_size):
     latest = w3.eth.getBlock('latest')
 
-    if latest['number'] == 0 or sample_size == 0:
-        return 0
-
     constrained_sample_size = min(sample_size, latest['number'])
+    if constrained_sample_size == 0:
+        raise ZeroDivisionError('Constrained sample size is 0')
```

Comment:
I think I'd use a `web3.exceptions.ValidationError` here.  Since this isn't actual division by zero, but rather pre-validation.